library/rtsp/src/main/java/com/google/android/exoplayer2/source/rtsp/RtspClient.java

```diff
@@ -124,6 +124,7 @@
   private final PlaybackEventListener playbackEventListener;
   private final String userAgent;
   private final boolean debugLoggingEnabled;
+  @Nullable private final SocketFactory socketFactory;
```

Comment:
The custom socketFactory is passed from `RtspMediaSource.Factory` right? IMO except for the Factory class, all the socketFactory field should be non-null. More specifically, the logic on L296 should be moved into the `RtspMediaSource.Factory.createMediaSource() method`.